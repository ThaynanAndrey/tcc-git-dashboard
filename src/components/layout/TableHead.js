import React from 'react';

/**
 * Stateless component with the function to presenting the table's head.
 * 
 * @author Thaynan Nunes
 */
const TableHead = ({ headsNames}) => {
    const thHeadNames = headsNames.map((name, index) => (<th key={index}>{ name }</th>));

    return (
        <thead>
            <tr>
                { thHeadNames }
            </tr>
        </thead>
    );
}

export default TableHead;