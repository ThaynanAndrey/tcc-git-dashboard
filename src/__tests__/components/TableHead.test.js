import React from 'react';
import renderer from 'react-test-renderer';

import TableHead from '../../components/layout/TableHead';

it('renders correctly when there are no headsNames', () => {
    const headsNames = [];
    const tree = renderer.create(
        <TableHead headsNames={headsNames} />
        ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders correctly when there are multiple headsNames', () => {
    const headsNames = ["Casa", "carro", ""];
    const tree = renderer.create(<TableHead headsNames={headsNames} />).toJSON();

    expect(tree).toMatchSnapshot();
});