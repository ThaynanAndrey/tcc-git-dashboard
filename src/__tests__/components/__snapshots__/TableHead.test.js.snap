// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are multiple headsNames 1`] = `
<thead>
  <tr>
    <th>
      Casa
    </th>
    <th>
      carro
    </th>
    <th>
      
    </th>
  </tr>
</thead>
`;

exports[`renders correctly when there are no headsNames 1`] = `
<thead>
  <tr />
</thead>
`;
